packages/dappmanager/src/calls/wifi.ts

```diff
@@ -0,0 +1,92 @@
+import { ComposeFileEditor } from "../modules/compose/editor";
+import { logContainer } from "../modules/docker/api";
+import { listContainer } from "../modules/docker/list";
+import { CurrentWifiCredentials, WifiReport } from "../types";
+import params from "../params";
+
+const WIFI_KEY_SSID = "SSID";
+const WIFI_KEY_PASSWORD = "WPA_PASSPHRASE";
+
+/**
+ * Return wifi report
+ */
+export async function wifiReportGet(): Promise<WifiReport> {
+  const wifiContainer = await listContainer({
+    containerName: params.wifiContainerName
+  });
+  if (!wifiContainer)
+    throw Error(
+      `Wifi is not present. Container name: ${params.wifiContainerName}`
+    );
+
+  // exitCode 0 means that wifi package was manually stopped. No errors
+  if (wifiContainer.exitCode === 0) wifiContainer.state === "paused";
+
+  let report;
+  let info = "";
+
+  switch (wifiContainer.state) {
+    case "created":
+      info = "Wifi has not been initialized yet";
+    case "running":
+      info = "Wifi is currently running";
+    case "restarting":
+      info = "Wifi restarting. Wait until it starts";
+    case "dead":
+      info =
+        "Wifi service dead, you must manually remove it and install it again";
+      report = {
+        lastLog: parseWifiLogs(await getWifiLastLog()),
+        exitCode: wifiContainer.exitCode
+      };
+    case "exited":
+      info = "Wifi service exited due to an internal error";
+      report = {
+        lastLog: parseWifiLogs(await getWifiLastLog()),
+        exitCode: wifiContainer.exitCode
+      };
+    case "paused":
+      info = "Wifi service is paused. Restart wifi to get back access again";
+  }
+
+  return {
+    wifiContainer,
+    info,
+    report
+  };
+}
+
+/**
+ * Returns wifi credentials
+ */
+export async function wifiCredentialsGet(): Promise<CurrentWifiCredentials> {
+  const composeWifi = new ComposeFileEditor(params.wifiDnpName, true);
+  const wifiService = composeWifi.services()[params.wifiDnpName];
+  const wifiEnvs = wifiService.getEnvs();
+  if (
+    typeof wifiEnvs[WIFI_KEY_SSID] === "undefined" ||
```

Comment:
You can do `wifiEnvs[WIFI_KEY_SSID] === undefined` which is much more typesafe. `typeof` should never be used unless in very specific scenarios